src/snowflake/snowpark/dataframe_writer.py

```diff
@@ -92,6 +95,26 @@ def mode(self, save_mode: str) -> "DataFrameWriter":
         self._save_mode = str_to_enum(save_mode.lower(), SaveMode, "`save_mode`")
         return self
 
+    def partition_by(self, expr: ColumnOrSqlExpr) -> "DataFrameWriter":
+        """Specifies an expression used to partition the unloaded table rows into separate files. It can be a
+        :class:`Column`, a column name, or a SQL expression.
+        """
+        self._partition_by = expr
+        return self
+
+    def option(self, key: str, value: Any) -> "DataFrameWriter":
+        """Depending on the ``file_format_type`` specified, you can include more format specific options.
+        Use the options documented in the `Format Type Options <https://docs.snowflake.com/en/sql-reference/sql/copy-into-location.html#format-type-options-formattypeoptions>`__.
+        """
+        aliased_key = get_aliased_option_name(key, WRITER_OPTIONS_ALIAS_MAP)
+        self._cur_options[aliased_key] = value
+        return self
```

Comment:
do we need to clean up old ones after execution or before setting new values? otherwise the options keeps accumulated? I'm thinking of the following case:

df.write.options(a=1) # this has option a = 1
df.write.options(b=2) # this would have option a= 1 and b = 2

Reply:
This is intentional to allow daisy chaining. We want to allow df.writer to be something like
```
df.writer.option("opt1", val1).option("opt2", val2)
```

note that df.reader already supports the same pattern. We are extending the same functionality to writer.

Reply:
I see, if this is an existing pattern than I'm good with it